projects/bootloader/test/test_config.c

```diff
@@ -0,0 +1,171 @@
+#include "bootloader_mcu.h"
+#include "config.h"
+#include "crc32.h"
+#include "flash.h"
+#include "log.h"
+#include "persist.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+extern StatusCode __real_persist_init(PersistStorage *persist, FlashPage page, void *blob,
+                                      size_t blob_size, bool overwrite);
```

Comment:
This is great, but you should definitely comment on it since it's a super obscure feature that most people won't know unless they've dug into how mocks work. Also, I'd like to keep this, `s_function_pointer_hook`, and the `persist_init` mock itself together since they form one sort of logical unit.